### ISSUE TITLE
Question about Connector development

### ISSUE THREAD

**Issue by wernerdaehn** [open-metadata/OpenMetadata#9349]
I watched the 1h video about developing custom connectors and I am confused still. The reason I am asking is I am considering adding an SAP ERP connector. This will be a larger task and hence I want to do that only if I can estimate the time it will take me upfront.

1. I did not get the point about installing metadata in the server. If I want to add a connection and normal table-like info, is that needed?
2. SAP does not have a Python library but a Java library. Hence I would like to implement the connector in Java. Can I do that using the Java SDK instead of Python, Schema, Code Generation, .. etc
3. Workflows I would like to validate my understanding. These are used to trigger the ingestion process and if I use Java, I would lose that. How big of an issue is that?

Thanks and sorry for asking these very basic questions

**Comment by harshach** [comment]
@wernerdaehn please join our slack https://slack.open-metadata.org , we can work closely and help you with it

**Comment by pmbrull** [comment]
hi @wernerdaehn, thanks for reaching out! Some thoughts around this:

1. Not fully following what you mean here, if you could please share some further details it would be great. As @harshach pointed out, we can followup the discussion on Slack to make it easier.
2. Looks like SAP ERP might have some REST APIs https://api.sap.com/package/erp/rest we can call from Python. The client is just a possible approach, but in the end we just need some way (whatever it is) to grab metadata from the source system.
3. All the ingestion process is designed and built with Python. The workflow definition (the class that know how to execute the ingestion - the steps to take), etc. are defined in the `openmetadata-ingestion` package. The Java side of OpenMetadata are the server APIs, and when talking about ingestion, that's our final endpoint to push the data. Also, the ingestion container is managed with Airflow, so the integration from a scheduling perspective of the workflows also happens with Python.

So, TLDR: is it possible to build a process that gets metadata and sends it to OpenMetadata using Java? Yes, and with any language at hand thanks to the APIs, but we would lose the Ingestion Framework definition that powers up the ingestion with Python, so I'd recommend going the Python route for simplicity

**Comment by pmbrull** [comment]
hi @wernerdaehn, hope the above made sense. I will be closing this issue if there are no further questions. If you still have any doubts about it you can reach out to us in Slack and we can give you some help.

Thanks

**Comment by wernerdaehn** [comment]
Yes, thank you. Given your feedback the probability of me starting that increased. Let me try to connect to Slack.
